From 9605f5410ba65946042bde19deccee8104c8219e Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Tue, 12 Apr 2022 15:44:46 +0300
Subject: [PATCH 19/20] rtc: abx80x: add support for am1805

ambiq 1805 rtc advertises part-number 1800, but has all the features of
1805 ... ... ... add capabilities of this chip.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 drivers/rtc/rtc-abx80x.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-abx80x.c b/drivers/rtc/rtc-abx80x.c
index 73830670a..68b05e905 100644
--- a/drivers/rtc/rtc-abx80x.c
+++ b/drivers/rtc/rtc-abx80x.c
@@ -88,7 +88,7 @@
 
 static u8 trickle_resistors[] = {0, 3, 6, 11};
 
-enum abx80x_chip {AB0801, AB0803, AB0804, AB0805,
+enum abx80x_chip {AB0801, AB0803, AB0804, AB0805, AM1805,
 	AB1801, AB1803, AB1804, AB1805, RV1805, ABX80X};
 
 struct abx80x_cap {
@@ -102,6 +102,7 @@ static struct abx80x_cap abx80x_caps[] = {
 	[AB0803] = {.pn = 0x0803},
 	[AB0804] = {.pn = 0x0804, .has_tc = true, .has_wdog = true},
 	[AB0805] = {.pn = 0x0805, .has_tc = true, .has_wdog = true},
+	[AM1805] = {.pn = 0x1800, .has_tc = true, .has_wdog = true},
 	[AB1801] = {.pn = 0x1801},
 	[AB1803] = {.pn = 0x1803},
 	[AB1804] = {.pn = 0x1804, .has_tc = true, .has_wdog = true},
@@ -857,6 +858,7 @@ static const struct i2c_device_id abx80x_id[] = {
 	{ "ab0803", AB0803 },
 	{ "ab0804", AB0804 },
 	{ "ab0805", AB0805 },
+	{ "am1805", AM1805 },
 	{ "ab1801", AB1801 },
 	{ "ab1803", AB1803 },
 	{ "ab1804", AB1804 },
-- 
2.25.1

